ibm_catalog.json

```diff
@@ -0,0 +1,250 @@
+            },
+            {
+              "key": "cloud_monitoring_access_key"
+            },
+            {
+              "key": "cloud_monitoring_add_cluster_name"
+            },
+            {
+              "key": "cloud_monitoring_agent_name"
+            },
+            {
+              "key": "cloud_monitoring_agent_namespace"
+            },
+            {
+              "key": "cloud_monitoring_agent_tags"
+            },
+            {
+              "key": "cloud_monitoring_agent_tolerations"
+            },
+            {
+              "key": "cloud_monitoring_endpoint_type"
+            },
+            {
+              "key": "cloud_monitoring_instance_region",
+              "display_name": "Region",
+              "custom_config": {
+                "type": "vpc_region",
+                "grouping": "deployment",
+                "original_grouping": "deployment",
+                "config_constraints": {
+                  "generationType": "2"
+                }
+              }
+            },
+            {
+              "key": "cloud_monitoring_metrics_filter"
+            },
+            {
+              "key": "cloud_monitoring_secret_name"
+            },
+            {
+              "key": "cluster_config_endpoint_type"
+            },
+            {
+              "key": "cluster_data"
+            },
+            {
+              "key": "cloud_logs_ingress_endpoint"
+            },
+            {
+              "key": "cloud_logs_ingress_port"
+            },
+            {
+              "key": "logs_agent_enabled"
+            },
+            {
+              "key": "logs_agent_additional_log_source_paths"
+            },
+            {
+              "key": "logs_agent_additional_metadata"
+            },
+            {
+              "key": "logs_agent_enable_scc"
+            },
+            {
+              "key": "logs_agent_exclude_log_source_paths"
+            },
+            {
+              "key": "logs_agent_iam_api_key"
+            },
+            {
+              "key": "logs_agent_iam_environment"
+            },
+            {
+              "key": "logs_agent_iam_mode",
+              "options": [
+                {
+                  "displayname": "Trusted Profile",
+                  "value": "TrustedProfile"
+                },
+                {
+                  "displayname": "IAM API Key",
+                  "value": "IAMAPIKey"
+                }
+              ]
+            },
+            {
+              "key": "logs_agent_log_source_namespaces"
+            },
+            {
+              "key": "logs_agent_name"
+            },
+            {
+              "key": "logs_agent_namespace"
+            },
+            {
+              "key": "logs_agent_selected_log_source_paths"
+            },
+            {
+              "key": "logs_agent_tolerations"
+            },
+            {
+              "key": "logs_agent_trusted_profile"
```

Comment:
since `logs_agent_iam_mode` default to `TrustedProfile`, that makes this variable required. So I would force the variable to required here as thats the default. User can always leave it at `null` and enter a value for apikey if they want. Ensure variable description makes that clear